sdk/identity/azure-identity/azure/identity/_credentials/cli_credential.py

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import platform
+_USE_SHELL = platform.system() == 'Windows'
+
+import json
+import time
+from datetime import datetime
+
+from subprocess import check_output, CalledProcessError, STDOUT
+import six
+
+from azure.core.credentials import AccessToken
+from azure.core.exceptions import ClientAuthenticationError
+
+
+def _microseconds_parsed(timestamp):
+    _index = timestamp.index('.')
+    _microseconds = timestamp[_index + 1:]
+    return '.'.join([timestamp[:_index]] + ['{:06d}'.format(int(_microseconds))])
```

Comment:
`strptime` doesn't require the microseconds be 6 characters wide.

Reply:
Is it possible we get for example "2020-01-01 10:00:00.1" in the CLI access token? This case `strptime` returns 100000μs for the microseconds part.

Reply:
I don't know how the CLI formats microseconds, but consider:
1. the CLI, being a Python program, probably uses a formatting method with the same behavior as `strptime`
2. the largest possible error is less than one second

The potential error is acceptable to me--we don't need microsecond resolution for access tokens.

Reply:
removed this method